wallet/src/main/scala/org/bitcoins/wallet/internal/RescanHandling.scala

```diff
@@ -205,14 +205,11 @@ private[wallet] trait RescanHandling extends WalletLogger {
   private def downloadAndProcessBlocks(
       blocks: Vector[DoubleSha256Digest]): Future[Unit] = {
     logger.info(s"Requesting ${blocks.size} block(s)")
-    blocks.foldLeft(Future.unit) { (prevF, blockHash) =>
-      val completedF = subscribeForBlockProcessingCompletionSignal(blockHash)
-      for {
-        _ <- prevF
-        _ <- nodeApi.downloadBlocks(Vector(blockHash))
-        _ <- completedF
-      } yield ()
-    }
+
+    val downloadF = nodeApi.downloadBlocks(blocks)
+    downloadF.failed.foreach(err =>
+      logger.error(s"Failed to download block hashes=$blocks", err))
+    downloadF
```

Comment:
this change can cause us to not process the blocks in order

Reply:
How so? You do seem to be right but i don't understand why. Is ordering not preserved when you are send the request block messages over to your bitcoind peer?

Reply:
the peer isn't guaranteed to send them to us in order and we when received a block we immediately process it, so when the peer sends the blocks, we will process them all concurrently

potential fix would be maybe change `OnBlockReceived` to `OnBlocksReceived` so you can process in order and not concurrently